### DISCUSSION TITLE
"ir repeater"

### DISCUSSION THREAD

**Discussion by fusebyte** [arendst/Tasmota#11985] [Q&A] [answered]
Hi, i'm nearly new in the tasmota world and i'm trying to make a custom firmware of tasmota for being able to resend the same message that ir has received.
I've tried for much time but i didn't find the way to solve this problem.
Thank u in advance.

**Comment by sfromis** [comment, marked as answer]
You can have a [rule](https://tasmota.github.io/docs/Rules/) issue the IRSend command, having the various values already in `%var1%` and so on, basically the same JSON format as you see in the IrReceived JSON payloads. To get the values, you can have a separate rule for each field, like 
`on IrReceived#Protocol do var1 %value% endon`

Doing it backend via MQTT may be a bit more straightforward....

**Reply by fusebyte** [reply]
i'd like to make something more stable modifyng the firmware

**Reply by sfromis** [reply]
Of course, with open source, choosing to modify the firmware instead is just a question of getting to work, after probably more than merely a little bit of learning curve. There is some [developer docs](https://tasmota.github.io/docs/For-Developers/), but far from covering it all.

**Comment by sciurius** [comment]
Does this also work for signals (protocols) that are not understood? In a live environment I have something like this:
````
{"IrReceived":{"Protocol":"RC6","Bits":20,"Data":"0x1000C","DataLSB":"0x800030","Repeat":0}}
{"IrReceived":{"Protocol":"UNKNOWN","Bits":17,"Hash":"0xCBD932BC","Repeat":0}}
{"IrReceived":{"Protocol":"UNKNOWN","Bits":10,"Hash":"0x249F6B34","Repeat":0}}
{"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0x482C609F","DataLSB":"0x123406F9","Repeat":0}}
{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1}}
{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1}}
{"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0x482C609F","DataLSB":"0x123406F9","Repeat":0}}
{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1}}
{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1}}
{"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0x482C40BF","DataLSB":"0x123402FD","Repeat":0}}
{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1}}
{"IrReceived":{"Protocol":"UNKNOWN","Bits":17,"Hash":"0x4AC72F16","Repeat":0}}
````
According to the docs the `"Hash"` can not be used to create messages.
